relayer/pkg/chainlink/ocr2/contract_cache.go

```diff
@@ -31,11 +30,11 @@ type contractCache struct {
 	stop, done chan struct{}
 
 	reader *contractReader
-	cfg    config.Config
+	cfg    Config
 	lggr   logger.Logger
 }
 
-func NewContractCache(cfg config.Config, reader *contractReader, lggr logger.Logger) *contractCache {
+func NewContractCache(cfg Config, reader *contractReader, lggr logger.Logger) *contractCache {
```

Comment:
Weird to depend on a local struct here `reader *contractReader` instead of a proper interface. We should probably define an `ocr2.Reader` interface, the `ContractCache` should require it to read, cache the data and also implement so it can be used as an `ocr2.Reader`.

Reply:
nice and clean fix:
```
type Reader interface {
	types.ContractConfigTracker
	median.MedianContract
}

var _ Reader = (*contractReader)(nil)
```